clarifai/cli/model.py

```diff
@@ -60,11 +81,49 @@ def test_locally(model_path, keep_env=False):
     type=click.Path(exists=True),
     required=True,
     help='Path to the model directory.')
-def run_locally(model_path):
-  """Run model locally and starts a GRPC server to serve the model."""
+@click.option(
+    '--port',
+    '-p',
+    type=int,
+    default=8000,
+    show_default=True,
+    help="The port to host the gRPC server for running the model locally. Defaults to 8000.")
+@click.option(
+    '--mode',
+    type=click.Choice(['env', 'container'], case_sensitive=False),
+    default='env',
+    show_default=True,
+    help=
+    'Specifies how to run the model: "env" for virtual environment or "container" for Docker container. Defaults to "env".'
+)
+@click.option(
+    '--keep_env',
+    is_flag=True,
+    help=
+    'Keep the virtual environment after testing the model locally (applicable for virtualenv mode). Defaults to False.'
+)
+@click.option(
+    '--keep_image',
+    is_flag=True,
+    help=
+    'Keep the Docker image after testing the model locally (applicable for container mode). Defaults to False.'
+)
+def run_locally(model_path, port, mode, keep_env, keep_image):
+  """Run the model locally and start a gRPC server to serve the model."""
   try:
     from clarifai.runners.models import model_run_locally
-    model_run_locally.main(model_path, run_model_server=True)
-    click.echo(f"Model server started locally from {model_path}.")
+
+    if mode == "virtualenv":
```

Comment:
I assume modes are `env` and `container` only, is this correct here to have `virtualenv`??

Reply:
right updated it